test/sql/catalog/test_temporary_schema.test

```diff
@@ -0,0 +1,455 @@
+# name: test/sql/catalog/test_temporary_schema.test
+# description: Test temporary objects in non-default schemas
+# group: [catalog]
+
+# =============================================================================
+# Basic temporary objects in custom schemas
+# =============================================================================
+
+# Create a persistent schema first
+statement ok
+CREATE SCHEMA myschema
+
+# Create temporary table in custom schema - schema should auto-create in temp catalog
+statement ok
+CREATE TEMPORARY TABLE myschema.t1(i INTEGER)
+
+# Insert and query
+statement ok
+INSERT INTO myschema.t1 VALUES (1), (2), (3)
+
+query I rowsort
+SELECT * FROM myschema.t1
+----
+1
+2
+3
+
+# Verify the table is in the temp catalog
+query IIII
+SELECT database_name, schema_name, table_name, temporary FROM duckdb_tables() WHERE table_name = 't1'
+----
+temp	myschema	t1	true
+
+# Verify the schema was auto-created in temp catalog
+query II
+SELECT database_name, schema_name FROM duckdb_schemas() WHERE schema_name = 'myschema' ORDER BY database_name
+----
+memory	myschema
+temp	myschema
+
+# Create temporary view in custom schema
+statement ok
+CREATE TEMPORARY VIEW myschema.v1 AS SELECT i * 2 AS doubled FROM myschema.t1
+
+query I rowsort
+SELECT * FROM myschema.v1
+----
+2
+4
+6
+
+# Create temporary sequence in custom schema
+statement ok
+CREATE TEMPORARY SEQUENCE myschema.seq1
+
+query I
+SELECT nextval('myschema.seq1')
+----
+1
+
+query I
+SELECT nextval('myschema.seq1')
+----
+2
+
+# Create temporary macro in custom schema
+statement ok
+CREATE TEMPORARY MACRO myschema.double(x) AS x * 2
+
+query I
+SELECT myschema.double(21)
+----
+42
+
+# =============================================================================
+# Auto-creation of schemas in temp catalog
+# =============================================================================
+
+# Create temporary table in a schema that doesn't exist anywhere yet
+statement ok
+CREATE TEMPORARY TABLE newschema.t1(x INTEGER)
+
+# The schema should be auto-created in temp catalog only
```

Comment:
What happens if we then create a non-temporary schema by this same name?
I imagine that should work but lookups are still shadowed.

Can we extend the test for this?
Followed by dropping the temporary counterpart to verify shadowing stops